llvm/include/llvm/Option/OptParser.td

```diff
@@ -93,6 +93,11 @@ class OptionGroup<string name> {
 
 // Define the option class.
 
+class HelpTextForVisibility<OptionVisibility visibility, string text> {
```

Comment:
Naming is hard :) How about `HelpTextVariant`? Basically, `HelpTextForVisibility` refers to the implementation detail ("visibility") and I suggest that the name reflects the functionality provided.

Reply:
I've changed it to:
HelpTextForVariants (for a single entry in the list, which is many variants to one string)
Help**Texts**ForVariants (for the list itself, which is many HelpTextForVariants)

Which is bound to confuse someone, but it's that or have them be confused why HelpTextForVariant is actually a many to one mapping which seems worse.